static/app/components/nav/utils.tsx

```diff
@@ -37,6 +37,10 @@ export interface NavSidebarItem extends NavItem {
    * The icon to render in the sidebar
    */
   icon: React.ReactElement;
+  /**
+   * A unique identifier string, used as a key for analytics
+   */
+  id: string;
```

Comment:
If there a reasonable chance (and from experience, I would probably say that there is), then I would rather namespace this under `analytics` object. It's not much of a difference, but it gives you more flexibility and isolation of the analytics and item concept (you'd just access it via `item.analytics.id`)

Reply:
On a separate note, you could apply inversion of control here, and define this as a callback function, so that if someone needs to extend anything, they don't need to go through the item definition and add fields, bloating the struct

Reply:
Changed `id` to `analyticsKey` in f1b3149dee8. I also like the IoC idea, will keep that in mind if the field becomes annoying down the road.